### ISSUE TITLE
cannot stop beacon node with CTRL+C: UnhandledPromiseRejectionWarning - Cannot read property 'unsubscribeFromNewEpoch'

### ISSUE THREAD

**Issue by q9f** [ChainSafe/lodestar#1297]
**Describe the bug**
CTRL+C does not allow me to exit the running beacon node

```
ict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 228)
(node:52111) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'unsubscribeFromNewEpoch' of undefined
    at ArrayDagLMDGHOST.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/chain/forkChoice/arrayDag/lmdGhost.js:218:16)
    at BeaconChain.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/chain/chain.js:167:27)
    at BeaconNode.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/node/nodejs.js:168:22)
    at async process.cleanup (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar-cli/lib/cmds/beacon/cmds/run/run.js:65:5)
(node:52111) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 230)
(node:52111) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'unsubscribeFromNewEpoch' of undefined
    at ArrayDagLMDGHOST.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/chain/forkChoice/arrayDag/lmdGhost.js:218:16)
    at BeaconChain.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/chain/chain.js:167:27)
    at BeaconNode.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/node/nodejs.js:168:22)
    at async process.cleanup (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar-cli/lib/cmds/beacon/cmds/run/run.js:65:5)
(node:52111) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 232)
(node:52111) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'unsubscribeFromNewEpoch' of undefined
    at ArrayDagLMDGHOST.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/chain/forkChoice/arrayDag/lmdGhost.js:218:16)
    at BeaconChain.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/chain/chain.js:167:27)
    at BeaconNode.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/node/nodejs.js:168:22)
    at async process.cleanup (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar-cli/lib/cmds/beacon/cmds/run/run.js:65:5)
(node:52111) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 234)
(node:52111) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'unsubscribeFromNewEpoch' of undefined
    at ArrayDagLMDGHOST.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/chain/forkChoice/arrayDag/lmdGhost.js:218:16)
    at BeaconChain.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/chain/chain.js:167:27)
    at BeaconNode.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/node/nodejs.js:168:22)
    at async process.cleanup (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar-cli/lib/cmds/beacon/cmds/run/run.js:65:5)
(node:52111) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 236)
(node:52111) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'unsubscribeFromNewEpoch' of undefined
    at ArrayDagLMDGHOST.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/chain/forkChoice/arrayDag/lmdGhost.js:218:16)
    at BeaconChain.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/chain/chain.js:167:27)
    at BeaconNode.stop (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar/lib/node/nodejs.js:168:22)
    at async process.cleanup (/usr/local/share/.config/yarn/global/node_modules/@chainsafe/lodestar-cli/lib/cmds/beacon/cmds/run/run.js:65:5)
(node:52111) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 238)
```

**Expected behavior**
CTRL+C should shut down the node

**Steps to Reproduce**
1. `node /usr/local/bin/lodestar beacon run --testnet medalla --eth1.enabled true --eth1.provider.url http://localhost:8545 --eth1.provider.network 5 --eth1.depositContract.deployedAt 3085928 --eth1.depositContract.address 0x07b39F4fDE4A38bACe212b546dAc87C58DfE3fDC`
2. CTRL+C

 - OS: Archlinux
 - Branch: `master`
 - Commit hash: 02d82b2d550c596fa91ced65194ccb9c33f58175

**Comment by dapplion** [comment]
Only usage of `unsubscribeFromNewEpoch` related to the stack trace above is this (below). But clock is checked before called, how can it be that is called despite being undefined?
```ts
export class ArrayDagLMDGHOST {
  public async stop(): Promise<void> {
    this.clock && this.clock.unsubscribeFromNewEpoch(this.onTick);
  }
  ...
```

**Comment by mpetrunic** [comment]
resolved via #1440 
It should allow force exit while we take some time to ensure everything is properly stopped
